forum/qa-plugin/tips-widget/tips-widget-admin.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class tips_widget_admin
+{
+	function option_default($option)
+	{
+		switch ($option)
```

Comment:
switch nie jest po to by generować X punktów wyjścia z metody. Od przerwania switcha jest break a nie return. Metoda nie ma prawa mieć więcej niż jednego returna

Reply:
Ale czemu tak sądzisz? Switch to zwykła konstrukcja sterująca i nie widzę żadnego problemu w używaniu w niej return.

Reply:
Chociażby literatura Bertranda Meyera i jego koncepcja programowania kontraktowego. Każda metoda ma w kontrakcie zapisane wejście i wyjście (poza voidem). Jest zawsze jedno wejście, które można w prosty sposób zwalidować oraz jedno wyjście, które jest równie proste do zweryfikowania. 
Dużo lepszym rozwiązaniem jest zarezerwowanie w pamięci przestrzeni tymczasowej tego, co ma być zwrócone. Jest mniej skoków, funkcja może być zapamiętana lepiej w opcache serwera.

Reply:
Poza tym możemy mieć 2 typy wyjścia z funkcji - zwrócenie wartości, lub wyjątku. Obie te rzeczy powinny być do obsłużenia. Używanie wielu returnów jest po prostu brzydkie. To jest podstawa w utrzymywaniu jakiejkolwiek jakości kodu